scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -713,11 +742,11 @@ object Execution {
      */
     override def zip[U](that: Execution[U]): Execution[(T, U)] =
       that match {
-        case WriteExecution(h, t, otherFn) =>
+        case WriteExecution(h, t, otherFn, tempFiles) =>
           val newFn = { (conf: Config, mode: Mode) =>
             (fn(conf, mode), otherFn(conf, mode))
           }
-          WriteExecution(head, h :: t ::: tail, newFn)
+          WriteExecution(head, h :: t ::: tail, newFn, tempFiles)
```

Comment:
don't you need to merge the sets from the left and right? Seems like we are losing `this.tempFilesToCleanup`.